cromwell_tools/cromwell_api.py

```diff
@@ -0,0 +1,439 @@
+               collection_name=None, label=None, validate_labels=False, on_hold=False):
+        # TODO: Allow more inputs files rather 2 to be consistent with the Cromwell API.
+        # TODO: Add `raise_for_status` flag to this method and remove the hard-coded `raise_for_status()` call.
+        # TODO: Purifying the methods by taking the specific retry policy out from this method.
+        # TODO: Add back `prepare_workflow_manifest` to simplify this method and allow file paths as inputs.
+        """ Submits a workflow to Cromwell.
+
+        Args:
+            auth (cromwell_tools.cromwell_auth.CromwellAuth): authentication class holding auth information to a Cromwell server.
+            wdl_file (_io.BytesIO): The workflow source file to submit for execution. From version 35,
+                Cromwell starts to accept URL to the WDL file besides actual WDL files.
+            inputs_file (_io.BytesIO): File-like object containing input data in JSON format.
+            options_file (Optional[_io.BytesIO]): Cromwell configs file.
+            inputs_file2 (Optional[_io.BytesIO]): Inputs file 2.
+            zip_file (Optional[_io.BytesIO]): Zip file containing dependencies.
+            collection_name (Optional[str]): Collection in SAM that the workflow should belong to.
+            label (Optional[Union[str, _io.BytesIO]]): JSON file containing a collection of
+                key/value pairs for workflow labels. # TODO: verify these types are accurate
+            validate_labels (Optional[bool]) If True, validate cromwell labels. (default False)
+            on_hold (Optional[bool]) Whether to submit the workflow in "On Hold" status.
+
+        Raises:
+            requests.exceptions.HTTPError: This will be raised when raise_for_status is True and Cromwell returns
+                a response that satisfies 400 <= response.status_code < 600.
+
+        Returns:
+            requests.Response: HTTP response from Cromwell.
+        """
+        if validate_labels and label is not None:
+            validate_cromwell_label(label)
+
+        files = {
+            'workflowSource': wdl_file,
+            'workflowInputs': inputs_file,
+            'workflowOnHold': json.dumps(on_hold)
+        }
+
+        if inputs_file2 is not None:
+            files['workflowInputs_2'] = inputs_file2
+        if zip_file is not None:
+            files['workflowDependencies'] = zip_file
+        if options_file is not None:
+            files['workflowOptions'] = options_file
+        if label:
+            files['labels'] = label
+        if collection_name:
+            files['collectionName'] = collection_name
+
+        response = requests.post(auth.url + cls._submit_endpoint, files=files, auth=auth.auth, headers=auth.header)
+        response.raise_for_status()
+        return response
+
+    @classmethod
+    def wait(cls, workflow_ids, timeout_minutes, auth, poll_interval_seconds=30, verbose=True):
```

Comment:
Please set a default for `timeout_minutes` here to match the default in the CLI

Reply:
Done